components/brave_today/browser/feed_controller.cc

```diff
@@ -270,42 +306,73 @@ void FeedController::OnPublishersUpdated(PublishersController* controller) {
-         const std::string& locale) {
-        // Handle the response
-        auto response_handler = base::BindOnce(
-            [](FeedController* controller, GetFeedItemsCallback callback,
-               api_request_helper::APIRequestResult api_request_result) {
-              std::string etag;
-              if (api_request_result.headers().contains(kEtagHeaderKey)) {
-                etag = api_request_result.headers().at(kEtagHeaderKey);
-              }
-              VLOG(1) << "Downloaded feed, status: "
-                      << api_request_result.response_code()
-                      << " etag: " << etag;
-              // Handle bad response
-              if (api_request_result.response_code() != 200 ||
-                  api_request_result.body().empty()) {
-                LOG(ERROR) << "Bad response from brave news feed.json. Status: "
-                           << api_request_result.response_code();
-                std::move(callback).Run({});
-                return;
-              }
-              // Only mark cache time of remote request if
-              // parsing was successful
-              controller->current_feed_etag_ = etag;
-              FeedItems feed_items;
-              ParseFeedItems(api_request_result.body(), &feed_items);
-              std::move(callback).Run(std::move(feed_items));
-            },
-            base::Unretained(controller), std::move(callback));
-        // Send the request
-        GURL feed_url(GetFeedUrl(locale));
-        VLOG(1) << "Making feed request to " << feed_url.spec();
-        controller->api_request_helper_->Request("GET", feed_url, "", "", true,
-                                                 std::move(response_handler),
-                                                 brave::private_cdn_headers);
+         Publishers publishers) {
+        auto locales = GetMinimalLocalesSet(
+            controller->channels_controller_->GetChannelLocales(), publishers);
+        VLOG(1) << "Going to fetch feed items for ["
+                << base::JoinString(locales, ", ") << "]";
+        auto locales_fetched_callback = base::BarrierCallback<FeedItems>(
+            locales.size(),
+            base::BindOnce(
+                [](GetFeedItemsCallback callback,
+                   std::vector<FeedItems> feed_items_unflat) {
+                  std::size_t total_size = 0;
+                  for (const auto& collection : feed_items_unflat)
+                    total_size += collection.size();
+
+                  FeedItems all_feed_items;
+                  all_feed_items.reserve(total_size);
+                  for (auto& collection : feed_items_unflat) {
+                    auto it = collection.begin();
+                    while (it != collection.end()) {
+                      all_feed_items.insert(all_feed_items.end(),
+                                            *std::make_move_iterator(it));
+                      it = collection.erase(it);
+                    }
```

Comment:
nit: I'm really not sure if this would work but can we use this for simple code?

```suggestion
                    all_feed_items.insert(all_feed_items.end(), 
                                          std::make_move_iterator(collection.begin()),
                                          std::make_move_iterator(collection.end()));
```

Reply:
Works nicely! I was blindly cargo culting a similar bit of code :D